### PR TITLE
SA1518 - The line endings at the end of a file do not match the setti…

### PR DESCRIPTION
**Pull request by elachlan**
Relates to #7174
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1518.md

The two remaining warnings/suggestions are from `RoslynImmutableInterlocked` pulled in from `Microsoft.CodeAnalysis.Collections` in `Microsoft.Build.Framework`.

Solution for the warnings from the nuget package is outlined in https://github.com/dotnet/roslyn/issues/55992

### CODE REVIEW

File: src/Build.OM.UnitTests/Construction/ProjectItemElement_Tests.cs [hunk 1]
```diff
@@ -1038,4 +1038,4 @@ private static ProjectItemElement GetItemFromContent(string content)
             return Helpers.GetFirst(project.Items);
         }
     }
-}
\ No newline at end of file
+}
```
Comment (Forgind) [inline src/Build.OM.UnitTests/Construction/ProjectItemElement_Tests.cs:1041]: SA 1518 even applies to newlines at the end of files? I'm not (personally) opposed to taking those out, but I imagine they'd come back in with various editors so probably not worth worrying about. Removing multiple blank lines at the end of a file seems more clear-cut to me.

Comment (elachlan) [reply src/Build.OM.UnitTests/Construction/ProjectItemElement_Tests.cs:1041]: The analyser uses the editor config settings for new lines. So I don't think they will get overridden by the editor later unless the settings change.

**Review by Forgind — APPROVED** [review]

**Comment by Nirmal4G** [comment]
This only affects the `cs` souces, right? Lot of other files have this issue too.

**Comment by elachlan** [comment]
> This only affects the `cs` souces, right? Lot of other files have this issue too.

The analyser has been applied to the whole solution.

**Comment by Nirmal4G** [comment]
No, only C# files have been touched.

**Comment by elachlan** [comment]
> No, only C# files have been touched.

This issue is specific to the stylecop analyser. I ran the specific analyser fix all for the solution. It only works against .net code and not anything else. Its not a solution wide analyser for all files.

**Comment by Nirmal4G** [comment]
That clears it up. I have a PR that touches the XAML, Props and Targets files. I was looking to see if I should rebase on top of yours but yours doesn't touch those files. So, no need.

**Comment by elachlan** [comment]
can you link it here? Are you using an Analyser so they don't sneak back in?

**Comment by sharwell** [comment]
It is not desirable for Microsoft.CodeAnalysis.Collections to omit the trailing newline, so feel free to send a Roslyn PR to just fix those two cases.

**Comment by Forgind** [comment]
Thanks everyone!

**Comment by elachlan** [comment]
Rolsyn files have been fixed. Awaiting a release and msbuild to update before we enable the rule.

**Pull request merged**